torch_frame/data/stats.py

```diff
@@ -160,6 +160,12 @@ def compute_col_stats(
 ) -> Dict[StatType, Any]:
     if stype == torch_frame.numerical:
         ser = ser.mask(ser.isin([np.inf, -np.inf]), np.nan)
+        try:
+            np.isfinite(ser) | ser.isna()
+        except TypeError:
+            raise TypeError("Numerical series contains invalid entries."
+                            "Please make sure your numerical series "
+                            "contains only numerical values or nans.")
```

Comment:
Can't we use https://pandas.pydata.org/docs/reference/api/pandas.api.types.is_numeric_dtype.html?